pw-3/src/4-task.ts

```diff
@@ -0,0 +1,35 @@
+type NumberParametersFunction = (...numbers: number[]) => number;
+
+function add(...numbers: number[]) {
+  let sum = 0;
+
+  for (const number of numbers) {
+    sum += number;
+  }
+
+  return sum;
+}
+
+function cacheWrapper(func: NumberParametersFunction): NumberParametersFunction {
```

Comment:
cache should be independent of number[]. It's a generic function

Reply:
Made cacheWrapper generic